collectd/files/interface.conf

```diff
@@ -1,8 +1,16 @@
+{%- from "collectd/map.jinja" import collectd_settings with context %}
+{%- set interface_settings = collectd_settings.plugins.interface -%}
+
+#
+# DO NOT EDIT
+#
+# This file is managed by salt via {{ source }}
+# Modify the config that generates this file instead
+#
+
 LoadPlugin interface
 
 <Plugin "interface">
-{% for interface in interfaces %}
-    Interface "{{ interface }}"
-{% endfor %}
-  IgnoreSelected {{ IgnoreSelected }}
+  Interface "{{ interface_settings.interface }}"
```

Comment:
It was never there before! In *this* PR I simply tried to change the style in which options are handled without making changes to the functionality or defaults.

In the case of interfaces we find:

```
- defaults:
        interfaces: {{ salt['pillar.get']('collectd:plugins:interface:interface', ['eth0']) }}
        IgnoreSelected: {{ salt['pillar.get']('collectd:plugins:interface:IgnoreSelected', 'false') }}
```
in https://github.com/saltstack-formulas/collectd-formula/blob/master/collectd/interface.sls#L15 which does *not* reference `collectd:interface:interfaces` and therefore *always* falls back to `['eth0']`.

I will fix inconsistencies and general snafu later, but tried not to change the way the formula behaves.